mlforecast/distributed/forecast.py

```diff
@@ -191,6 +205,19 @@ def _preprocess_partitions(
         window_info: Optional[WindowInfo] = None,
         fit_ts_only: bool = False,
     ) -> List[Any]:
+        if self.num_partitions:
+            partition = dict(by=id_col, num=self.num_partitions, algo="coarse")
```

Comment:
I couldn't find what the `algo='coarse'` does. I think the default (hash) is probably fine. Is there a reference you can share to take a look?

Reply:
That was a little bit tricky because I started playing around with just `dict(by=id_col, num=self.num_partitions)` and I found unexpected (for me) behavior. For example, in the following test, I expected to end up with just only two rows but I got three rows instead (same for spark):

![image](https://user-images.githubusercontent.com/10517170/229966584-8d16e63e-bad2-4c4c-9a64-3ca3fecc6d21.png)

After discussing it with the fugue guys, they suggested using `algo='coarse'` and it worked,

![image](https://user-images.githubusercontent.com/10517170/229966808-ee7e1d1a-aa88-490f-88a3-47c2dbfd64a7.png)

Basically, `num=2` (without `algo='coarse'`)  creates two partitions based on `id`  but inside those partitions, `get_n_ids` is applied to each of the ids.

Reply:
Got it, that seems better indeed. Thanks for the explanation!